crates/lib3h/tests/test_suites/mirror.rs

```diff
@@ -0,0 +1,121 @@
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+// Tests
+//--------------------------------------------------------------------------------------------------
+
+/// Empty function that triggers the test suite
+#[allow(dead_code)]
+fn test_setup_only(_nodes: &mut Vec<NodeMock>) {
+    // n/a
+}
+
+fn test_mirror_from_center(nodes: &mut Vec<NodeMock>) {
+    let entry = {
+        let mut node0 = nodes.remove(0);
+        let mut node1 = nodes.remove(0);
+        // node0 publishes data on the network
+        let entry = node0
+            .author_entry(&ENTRY_ADDRESS_1, vec![ASPECT_CONTENT_1.clone()], true)
+            .unwrap();
+
+        let expected = "HandleStoreEntryAspect\\(StoreEntryAspectData \\{ request_id: \"[\\w\\d_~]+\", space_address: HashString\\(\"\\w+\"\\), provider_agent_id: HashString\\(\"mirror_node0\"\\), entry_address: HashString\\(\"entry_addr_1\"\\), entry_aspect: EntryAspectData \\{ aspect_address: HashString\\(\"[\\w\\d]+\"\\), type_hint: \"NodeMock\", aspect: \"hello-1\", publish_ts: \\d+ \\} \\}\\)";
+
+        let _results = assert2_msg_matches!(node0, node1, expected);
+
+        assert_eq!(entry, node0.get_entry(&ENTRY_ADDRESS_1).unwrap());
+        nodes.insert(0, node1);
+        nodes.insert(0, node0);
+        entry
+    };
+
+    process_nodes(nodes);
+
+    for node in nodes {
+        println!("checking if {} has entry...", node.name());
+        assert_eq!(entry, node.get_entry(&ENTRY_ADDRESS_1).unwrap());
+        println!("yes!");
+    }
+}
+
+fn test_mirror_from_edge(nodes: &mut Vec<NodeMock>) {
+    let entry = {
+        let mut node0 = nodes.remove(0);
+        let mut noden = nodes.pop().unwrap();
+        // node0 publishes data on the network
+        let entry = noden
+            .author_entry(&ENTRY_ADDRESS_1, vec![ASPECT_CONTENT_1.clone()], true)
+            .unwrap();
+
+        let expected = "HandleStoreEntryAspect\\(StoreEntryAspectData \\{ request_id: \"[\\w\\d_~]+\", space_address: HashString\\(\"\\w+\"\\), provider_agent_id: HashString\\(\"mirror_node0\"\\), entry_address: HashString\\(\"entry_addr_1\"\\), entry_aspect: EntryAspectData \\{ aspect_address: HashString\\(\"[\\w\\d]+\"\\), type_hint: \"NodeMock\", aspect: \"hello-1\", publish_ts: \\d+ \\} \\}\\)";
+
+        let _results = assert2_msg_matches!(node0, noden, expected);
+
+        assert_eq!(entry, node0.get_entry(&ENTRY_ADDRESS_1).unwrap());
+        nodes.push(noden);
+        nodes.insert(0, node0);
+        entry
+    };
+
+    process_nodes(nodes);
+
+    for node in nodes {
+        println!("checking if {} has entry...", node.name());
+        assert_eq!(entry, node.get_entry(&ENTRY_ADDRESS_1).unwrap());
+        println!("yes!");
+    }
+}
+
+fn process_nodes(nodes: &mut Vec<NodeMock>) {
+    for _i in 0..50 {
+        process_nodes_inner(nodes);
+    }
+}
+fn process_nodes_inner(nodes: &mut Vec<NodeMock>) {
```

Comment:
I would like to tackle baking this into the processor harness eventually but it's fine for this PR. Nice trick using a private function to fix scoping issues.